src/cmstats.c

```diff
@@ -49,6 +50,25 @@ s_duplicator (const void *self)
     return (void*) fty_proto_dup ((fty_proto_t*) self);
 }
 
+static double
+safe_atof (const char *str)
+{
+    char *endptr;
+    errno = 0;
+    double ret = strtod (str, &endptr);
+    // we could not process whole str
+    if (ret == 0.0 && endptr != NULL && endptr[0] != '\0') {
```

Comment:
Technically, an input string like `0.000000abc` would also cause this - with the number part really being zero. But in our use-case it should not happen anyway. :)